### PR TITLE
(Revival) Bikehorn goes off if you step on it

### PR DESCRIPTION
**Pull request by ghost**
Revival of #26514, added delay as requested.
:cl:
 * rscadd: Bikehorns now go off when you step on them.

### CODE REVIEW

**Comment by Eneocho** [comment]
Does this include the syndie experimental bike horn?

**Comment by jwhitak** [comment]
It should, the syndie version is a subset of the main bike horn

File: code/game/objects/items/weapons/clown_items.dm [hunk 1]
```diff
@@ -59,6 +60,11 @@
 	if(honk())
 		add_fingerprint(user)
 
+/obj/item/weapon/bikehorn/Crossed(var/mob/living/AM)
```
Comment (Exxion) [inline code/game/objects/items/weapons/clown_items.dm:63]: You're typing the arg as `/mob/living` but not checking that it actually is that type, so it still has the problem of honking when anything at all moves over it

File: code/game/objects/items/weapons/clown_items.dm [hunk 2]
```diff
@@ -59,6 +60,11 @@
...
+	if (world.time > next_honk)
+		honk()
+		next_honk = world.time + 2 SECONDS
```
Comment (Exxion) [inline code/game/objects/items/weapons/clown_items.dm]: I'd recommend making the delay a variable

**Comment by Konater** [comment]
I can't code but I thought it already had a delay making shifty's request pointless

```
/obj/item/weapon/bikehorn/proc/honk()
	if(world.time - last_honk_time >= honk_delay)
		last_honk_time = world.time
		playsound(src, hitsound, 50, vary_pitch)
		return 1
	return 0
```

File: code/game/objects/items/weapons/clown_items.dm (hunk 1 above)
Comment (Konater) [reply code/game/objects/items/weapons/clown_items.dm:63]: I copied it from glass shards lol

Comment (Exxion) [reply code/game/objects/items/weapons/clown_items.dm:63]: Yes, this line, but not the rest of it

Comment (ghost) [reply code/game/objects/items/weapons/clown_items.dm:63]: Added that check, done

File: code/game/objects/items/weapons/clown_items.dm (hunk 2 above)
Comment (ghost) [reply code/game/objects/items/weapons/clown_items.dm]: Done, used honk_delay

**Pull request merged**